### ISSUE TITLE
thirdPartyResourceDataMapper needs to use KindToResource

### ISSUE THREAD

**Issue by liggitt** [kubernetes/kubernetes#17435]
it is currently hardcoding an incorrect conversion of kind to resource:
```
 func (t *thirdPartyResourceDataMapper) isThirdPartyResource(resource string) bool {
 	return resource == strings.ToLower(t.kind)+"s"
 }
```

**Comment by lavalamp** [comment]
Will need to be fixed before we turn this on.

**Comment by brendandburns** [comment]
Closed by #18131
